uwblueprint/sistering
# Clean shift request type names

## Ticket link

Closes #276


## Implementation description
* Updated naming of ShiftDataWithoutPostingId type
* `updatePosting` should now likewise include a shifts update
* Further shift creation conflict check by ensuring no redundant shifts are created


## Steps to test
1. Run this locally since backend changes don't apply to preview
2. Ensure that createPosting and updatePosting are still working
3. for create posting, go to the posting creation page to make sure it is still working
4. For the updatePosting mutation, make a call to it on the graphql playground and ensure it is still working w/o errors and shifts are properly overwrited

Testing mutation
```graphql
mutation {
  createPosting(posting: {
    title: "Test",
    type: INDIVIDUAL,
    status: PUBLISHED,
    description: "{\"blocks\":[{\"key\":\"ena93\",\"text\":\"asdasdasdasdasdad\",\"type\":\"unstyled\",\"depth\":0,\"inlineStyleRanges\":[],\"entityRanges\":[],\"data\":{}}],\"entityMap\":{}}",
    startDate: "2022-07-06",
    endDate: "2022-07-13",
    autoClosingDate: "2022-07-07",
    numVolunteers: 5,
    times: [
      {
        startTime: "2022-07-06T11:30",
        endTime: "2022-07-06T13:30"
      },
    ],
    recurrenceInterval: WEEKLY,
    branchId: "1",
    skills: [
      "1"
    ],
    employees: [
      "2"
    ]
  }) {
    id
  }
}
```


## What should reviewers focus on?
* Correctness and cleanliness

## Checklist
